src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -348,3 +352,30 @@ export const checkFormIsEncryptMode = (
     ? ok(form)
     : err(new ResponseModeError(ResponseMode.Encrypt, form.responseMode))
 }
+
+/**
+ * Creates an encrypted submission without saving it to the database.
+ * @param form Document of the form being submitted
+ * @param encryptedContent Encrypted content of submission
+ * @param version Encryption version
+ * @param attachmentMetadata
+ * @param verifiedContent Verified content included in submission, e.g. SingPass ID
+ * @returns Encrypted submission document which has not been saved to database
+ */
+export const createEncryptSubmissionWithoutSave = ({
+  form,
+  encryptedContent,
+  version,
+  attachmentMetadata,
+  verifiedContent,
+}: SaveEncryptSubmissionParams): IEncryptedSubmissionSchema => {
+  return new EncryptSubmissionModel({
```

Comment:
nit: should we use EncryptSubmissionModel.create instead for type checking?
(or does the new mongoose update now add type safety for this style of instantiation too?)

Reply:
the point of this function is _not_ to save the submission to the database, since this is a preview submission (hence the `WithoutSave` in the function name hahaha

Reply:
ah, missed it